rubem/modules/_interception.py

```diff
@@ -37,6 +39,9 @@ def srCalc(NDVI):
     :returns: Reflectances Simple Ratio (SR) [-]
     :rtype: float
     """
+    if not -1.0 <= NDVI <= 1.0:
```

Comment:
This implementation of the valid value range check for NDVI does not consider that the argument is a raster map in PCRaster format. To do this it should look like this:

```python
import pcraster as pcr
from pcraster.framework import generalfunctions

from validation._exception_validation import ValidationException

def srCalc(NDVI: pcr._pcraster.Field) -> pcr._pcraster.Field:
    max_value = generalfunctions.getCellValue(pcr.mapmaximum(NDVI), 1, 1)
    min_value = generalfunctions.getCellValue(pcr.mapminimum(NDVI), 1, 1)
    if not (max_value <= 1.0 and min_value >= -1.0):
        raise ValidationException(
            f"NDVI must be in the value range [-1.0, 1.0],"
            "value range was [{min_value}, {max_value}]"
        )
    SR = (1 + NDVI) / (1 - NDVI)
    return SR
```

**However, after further analysis of all the source code, I believe that checking the NDVI maps should be done right after reading them and not in every function that uses them.**

Checks within functions that implement formulas should be limited to addressing their boundary conditions and other constraints.